hexkit/custom_types.py

```diff
@@ -51,4 +73,4 @@
 #     def construct(cls, foo: str): ...
 # ```
 # Thus using a type alias for now:
-ContextConstructable = Any
+AsyncContextConstructable = Any
```

Comment:
That's indeed hard to type properly because the contextlib really doesn't make it easy. This is the best I could come up with: 

```python
class AsyncContextConstructable(Protocol):
    """A class that can be asynchronously constructed with a context manager.

    An AsyncContextConstructable is a class with a classmethod `construct`
    that creates an async context manager for safely setting up and tearing down
    an instance of that class.
    """

    construct: Callable
```

Reply:
I already tried this some weeks ago but mypy is not recognizing this.
Will push a proof in a minute.

Since we are already making checks for the correct type at runtime, I think there is little value in investing much more time into that right now. If you agree, I can put it on the refactoring list.

Reply:
Please have a look at this branch: https://github.com/ghga-de/hexkit/pull/36
Where mypy is reporting the following error: https://github.com/ghga-de/hexkit/actions/runs/3248759888/jobs/5330354785#step:6:169

Reply:
You can define `ValidAsyncConstructable` as a subclass of `AsyncConstructable` to make it work.